pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -2011,3 +2017,15 @@ type ScrapeClass struct {
 	// +optional
 	AttachMetadata *AttachMetadata `json:"attachMetadata,omitempty"`
 }
+
+// OTLPConfig is the configuration for writing to the OTLP endpoint.
+//
+// +k8s:openapi-gen=true
+type OTLPConfig struct {
+	// List of OpenTelemetry Attributes that should be promoted to metric labels, defaults to none.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	// +optional
+	PromoteResourceAttributes []string `json:"promoteResourceAttributes,omitempty"`
```

Comment:
We need a length validation on each item because Prometheus forbids empty attributes.

Reply:
Thanks. Got it .
Add `+kubebuilder:validation:Items:MinLength=1`  validation

Reply:
Thanks, today I learn about `+kubebuilder:validation:items:...` :)